.github/scripts/update_commit_hashes.py

```diff
@@ -0,0 +1,115 @@
+REQUEST_HEADERS = {
+    "Accept": "application/vnd.github.v3+json",
+    "Authorization": f"token {MERGEBOT_TOKEN}",
+}
+
+
+def git_api(url: str, params: Dict[str, str], post: bool = False) -> Any:
+    if post:
+        return requests.post(
+            f"https://api.github.com{url}",
+            data=json.dumps(params),
+            headers=REQUEST_HEADERS,
+        ).json()
+    else:
+        return requests.get(
+            f"https://api.github.com{url}",
+            params=params,
+            headers=REQUEST_HEADERS,
+        ).json()
+
+
+def parse_args() -> Any:
+    from argparse import ArgumentParser
+
+    parser = ArgumentParser("Rebase PR into branch")
+    parser.add_argument("--repo-name", type=str)
+    parser.add_argument("--branch", type=str)
+    return parser.parse_args()
+
+
+def make_pr(repo_name: str, branch_name: str) -> Any:
+    params = {
+        "title": f"[{repo_name} hash update] update the pinned {repo_name} hash",
+        "head": branch_name,
+        "base": "master",
+        "body": "This PR is auto-generated nightly by [this action](https://github.com/pytorch/pytorch/blob/master/" +
+        f".github/workflows/_update-commit-hash.yml).\nUpdate the pinned {repo_name} hash.",
+    }
+    response = git_api(f"/repos/{OWNER}/{REPO}/pulls", params, post=True)
+    print(f"made pr {response['number']}")
+    return response["number"]
+
+
+def approve_pr(pr_number: str) -> None:
+    params = {"event": "APPROVE"}
+    # use pytorchbot to approve the pr
+    REQUEST_HEADERS["Authorization"] = f"token {PYTORCHBOT_TOKEN}"
+    response = git_api(f"/repos/{OWNER}/{REPO}/pulls/{pr_number}/reviews", params, post=True)
+    REQUEST_HEADERS["Authorization"] = f"token {MERGEBOT_TOKEN}"
+
+
+def make_comment(pr_number: str) -> None:
+    params = {"body": "@pytorchbot merge -g"}
+    git_api(f"/repos/{OWNER}/{REPO}/issues/{pr_number}/comments", params, post=True)
+
+
+def main() -> None:
+    args = parse_args()
+
+    branch_name = os.environ["NEW_BRANCH_NAME"]
+    pr_num = None
+
+    # query to see if a pr already exists
+    params = {
+        "q": f"is:pr is:open in:title author:pytorchmergebot repo:pytorch/pytorch {args.repo_name} hash update"
+    }
+    response = git_api("/search/issues", params)
+    if response["total_count"] != 0:
+        # pr does exist
+        pr_num = response["items"][0]["number"]
+        response = git_api(f"/repos/{OWNER}/{REPO}/pulls/{pr_num}", {})
+        branch_name = response["head"]["ref"]
+        print(f"pr does exist, number is {pr_num}, branch name is {branch_name}")
+
+    # update file
+    hash = subprocess.run(
+        f"git rev-parse {args.branch}".split(), capture_output=True, cwd=f'{args.repo_name}'
+    ).stdout.decode("utf-8")
+    with open(f".github/{args.repo_name}_commit_hash.txt", "w") as f:
+        f.write(hash.strip())
+    if (
+        subprocess.run(
+            f"git diff --exit-code .github/{args.repo_name}_commit_hash.txt".split()
```

Comment:
Might be good to split this out into a separate variable and name what we're looking for just to make it more clear what this is supposed to be doing.